src/Client/AI/AIToolExecutionDisplay.h

```diff
@@ -0,0 +1,166 @@
+#pragma once
+
+#include <algorithm>
+#include <chrono>
+#include <string>
+#include <thread>
+#include <base/terminalColors.h>
+#include <ostream>
+#include <Common/StringUtils.h>
+#include <IO/ReadBufferFromString.h>
+#include <IO/ReadHelpers.h>
+
+namespace DB
+{
+
+class AIToolExecutionDisplay
+{
+public:
+    explicit AIToolExecutionDisplay(std::ostream & output_stream_, bool enable_colors = true)
+        : output_stream(output_stream_), use_colors(enable_colors)
+    {
+    }
+
+    void showThinking() const
+    {
+        if (use_colors)
+            output_stream << "\033[36m"; // Cyan
```

Comment:
move color codes to constants or see if there is another way this is done in the codebase.